lib/dec/avx2.c

```diff
@@ -36,11 +36,11 @@ while (srclen >= 45)
 
 	/* Check if all bytes have been classified; else fall back on bytewise code
 	 * to do error checking and reporting: */
-	if ((unsigned int)_mm256_movemask_epi8(s1mask | s2mask | s3mask | s4mask | s5mask) != 0xFFFFFFFF) {
+	if ((unsigned int)_mm256_movemask_epi8(_mm256_or_si256(_mm256_or_si256(_mm256_or_si256(_mm256_or_si256(s1mask, s2mask), s3mask), s4mask), s5mask)) != 0xFFFFFFFF) {
```

Comment:
Yeah, this is what I came up with as well, at first. I don't know if this is an improvement. Maybe something with a temporary variable like:

```c
__mm256 ormask;
ormask = _mm256_or_si256(s1mask, s2mask);
ormask = _mm256_or_si256(ormask, s3mask);
ormask = _mm256_or_si256(ormask, s4mask);
ormask = _mm256_or_si256(ormask, s5mask);

if ((unsigned int)_mm256_movemask_epi8(ormask) != 0xFFFFFFFF) {
```

The compiler will (probably) generate the exact same code. Temporaries are "free" in modern compilers.